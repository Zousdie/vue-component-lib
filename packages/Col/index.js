import sfc from '../utils/sfc';

const { def, bem } = sfc('col');

export default def({
  props: {
    tag: {
      type: String,
      default: 'div',
    },
    span: {
      type: Number,
      default: 12,
    },
    offset: Number,
    xs: [Number, Object],
    sm: [Number, Object],
    md: [Number, Object],
    lg: [Number, Object],
    xl: [Number, Object],
    xll: [Number, Object],
  },

  render(h) {
    const classes = [bem()];

    const { gutter } = this.$parent;
    const pd = gutter ? `${gutter / 2}px` : null;

    ['span', 'offset'].forEach((prop) => {
      if (this[prop] || this[prop] === 0) {
        classes.push(
          prop !== 'span'
            ? `v-col-${prop}-${this[prop]}`
            : `v-col-${this[prop]}`,
        );
      }
    });

    ['xs', 'sm', 'md', 'lg', 'xl', 'xll'].forEach((size) => {
      const sz = this[size];
      if (typeof sz === 'number') {
        classes.push(`v-col-${size}-${sz}`);
      } else if (typeof sz === 'object') {
        const props = sz;
        Object.keys(props).forEach((prop) => {
          classes.push(
            prop !== 'span'
              ? `v-col-${size}-${prop}-${props[prop]}`
              : `v-col-${size}-${props[prop]}`,
          );
        });
      }
    });

    return h(this.tag, {
      class: classes,
      style: {
        paddingLeft: pd,
        paddingRight: pd,
      },
    }, this.$slots.default);
  },
});
