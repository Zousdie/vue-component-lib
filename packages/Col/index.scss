@import "../style/global";

$max-span: 12;
$res-list: ("xs", "sm", "md", "lg", "xl", "xll");

@include b(col) {
  box-sizing: border-box;
  float: left;

  &-0 {
    display: none;
  }

  @for $i from 0 through $max-span {
    &-#{$i} {
      width: (1 / $max-span * $i * 100) * 1%;
    }

    &-offset-#{$i} {
      margin-left: (1 / $max-span * $i * 100) * 1%;
    }
  }

  @each $item in $res-list {
    @include res($item) {
      &-#{$item}-0 {
        display: none;
      }

      @for $i from 0 through $max-span {
        &-#{$item}-#{$i} {
          width: (1 / $max-span * $i * 100) * 1%;
        }

        &-#{$item}-offset-#{$i} {
          margin-left: (1 / $max-span * $i * 100) * 1%;
        }
      }
    }
  }
}
