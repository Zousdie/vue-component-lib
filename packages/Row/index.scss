@import "../style/global";

@include b(row) {
  box-sizing: border-box;

  @include clear;

  @include m(flex) {
    display: flex;
    flex-wrap: wrap;

    @include b(col) {
      flex: auto;
    }
  }
}
