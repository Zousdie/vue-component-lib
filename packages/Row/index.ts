import Vue from 'vue';
import sfc from '../utils/sfc';

export interface IRowComponent {
  Props: {
    tag?: string
    gutter?: number
    flex?: boolean
  }
  Slots: {
    default: ScopedSlot
  }
}

export type RowStyles = {
  marginLeft?: string,
  marginRight?: string,
}

const { def, bem } = sfc<IRowComponent>('row');

export default def({
  props: {
    tag: {
      type: String,
      default: 'div',
    },
    gutter: {
      type: Number,
      default: 0,
    },
    flex: {
      type: Boolean,
      default: false,
    },
  },

  render(h) {
    const style: RowStyles = {};
    if (this.gutter > 0) {
      style.marginLeft = `-${this.gutter / 2}px`;
      style.marginRight = style.marginLeft;
    }

    return h(this.tag, {
      class: bem({ flex: this.flex }),
      style,
    }, this.$slots.default);
  },
});
