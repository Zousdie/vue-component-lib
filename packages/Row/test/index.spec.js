import { mount } from '@vue/test-utils';
import Row from '..';
import Col from '../../Col';

describe('Component render', () => {
  it('Snapshot', () => {
    const wrapper = mount({
      components: {
        Row,
        Col,
      },
      template: `
        <div>
          <Row :gutter="10">
            <Col>
              all col
            </Col>
          </Row>
          <Row>
            <Col :span="4">
              span 4
            </Col>
            <Col :span="4" :offset="6">
              span 4 and offset 6
            </Col>
          </Row>
          <Row>
            <Col :xll="4">
              xll 4
            </Col>
          </Row>
          <Row>
            <Col :xll="{span:4, offset: 2}">
              xll obj 4 and offset 2
            </Col>
            <Col :xll="{span:4, offset: 2}">
              xll obj 4 and offset 2
            </Col>
          </Row>
        </div>
      `,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('flex mode', () => {
    const wrapper = mount({
      components: {
        Row,
        Col,
      },
      template: `
        <Row flex>
          <Col>flex mode</Col>
        </Row>
      `,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
