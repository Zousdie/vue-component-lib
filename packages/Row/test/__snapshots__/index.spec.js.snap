// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render Snapshot 1`] = `
<div>
  <div
    class="v-row"
    style="margin-left: -5px; margin-right: -5px;"
  >
    <div
      class="v-col v-col-12"
      style="padding-left: 5px; padding-right: 5px;"
    >
      
              all col
            
    </div>
  </div>
   
  <div
    class="v-row"
  >
    <div
      class="v-col v-col-4"
    >
      
              span 4
            
    </div>
     
    <div
      class="v-col v-col-4 v-col-offset-6"
    >
      
              span 4 and offset 6
            
    </div>
  </div>
   
  <div
    class="v-row"
  >
    <div
      class="v-col v-col-12 v-col-xll-4"
    >
      
              xll 4
            
    </div>
  </div>
   
  <div
    class="v-row"
  >
    <div
      class="v-col v-col-12 v-col-xll-4 v-col-xll-offset-2"
    >
      
              xll obj 4 and offset 2
            
    </div>
     
    <div
      class="v-col v-col-12 v-col-xll-4 v-col-xll-offset-2"
    >
      
              xll obj 4 and offset 2
            
    </div>
  </div>
</div>
`;

exports[`Component render flex mode 1`] = `
<div
  class="v-row v-row--flex"
>
  <div
    class="v-col v-col-12"
  >
    flex mode
  </div>
</div>
`;
